src/content/reference/react/index.md

```diff
@@ -10,6 +10,14 @@ This section provides detailed reference documentation for working with React. F
 
 The React reference documentation is broken down into functional subsections:
 
+## Rules of React {/*rules-of-react*/}
+
+JavaScript rendering libraries and frameworks like React have constraints or "rules" to make the programming model cohesive and easy to reason about, while also helping you prevent bugs in your code. The rules also have the added benefit of creating a safe space for React to optimize and run your code more efficiently.
+
+* [Components and Hooks must be pure](/reference/rules/components-and-hooks-must-be-pure) – Purity makes your code easier to understand, debug, and also allows the compiler to optimize your components and hooks correctly.
```

Comment:
I would recommend reducing the emphasis on the compiler, and focusing more heavily on correctness. Purity is foremost about correctness — it avoids bugs by design, which then in turns makes applications easier to understand and debug. I would maybe mention that purity allows React to automatically optimize your applications, not necessarily call out the compiler.

If we do mention "compiler", it should be "the upcoming React Compiler" (which is tedious, hence maybe just talk about potential for optimization instead)